client/interfaces.go

```diff
@@ -698,3 +725,15 @@ type InstanceFileResponse struct {
 	// If a directory, the list of files inside it
 	Entries []string
 }
+
+// GetPermissionsArgs is used in the call to GetPermissions to specify filtering behaviour.
+type GetPermissionsArgs struct {
+	// EntityType is the type of entity to filter against.
+	// If left unspecified, permissions will be returned for all entity types.
+	EntityType entity.Type
```

Comment:
As entity.Type is now referenced in the `shared/api` package directly, and I've not been able to find a precedent for the `shared/api` package referencing packages outside of itself, please can you move the `shared/entity` package to `shared/api/entity` to drive home that these constants are used in public API facing functions and cannot lightly/ever be changed.

Reply:
A bit like `shared/api/scriptlet`

Reply:
Ah OK so its just the client package not the shared/api package.  Makes sense thanks.